docs/_includes/errormessages.adoc

```diff
@@ -0,0 +1,274 @@
+|
+|<<user-abstract>>
+
+|WARNING
+|cannot retrieve contents of <x> at URI: <uri> (allow-uri-read attribute not enabled)
+|Reading from a URI is only allowed for some SAFE modes.
+|<<include-uri>>
+
+|WARNING
+|could not retrieve contents of <x> at URI: <uri>
+|Web address not found.
+|
+
+|WARNING
+|could not retrieve image data from URI: <uri>
+|Only occurs with `allow-uri-read` and `data-uri`. Check the URI.
+|<<include-images-by-full-url>>
+
+|WARNING
+|dropping line containing reference to missing attribute: <x>
+|
+|<<catch-a-missing-or-undefined-attribute>>
+
+|WARNING
+|image to embed not found or not readable: <file>
+|You used :data-uri: but the file could not be found.
+|<<managing-images>>
+
+|WARNING
+|include file not readable: <file>
+|File is not a text file.
+|<<include-nonasciidoc>>
+
+|WARNING
+|input path <file> is a <x>, not a file
+|File is not a text file.
+|<<include-nonasciidoc>>
+
+|WARNING
+|optional gem 'asciimath' is not installed. Functionality disabled.
+|asciimath is one of the libraries used for equations.
+|<<stem>>
+
+|WARNING
+|optional gem 'coderay' is not installed. Functionality disabled.
+|Coderay is used for source code highlighting.
+|<<coderay>>
+
+|WARNING
+|skipping reference to missing attribute: <x>
+|
+|<<catch-a-missing-or-undefined-attribute>>
+
+|WARNING
+|tables must have at least one body row
+|
+|<<tables>>
+
+|WARNING
+|tag '<x>' not found in include file: <file>
+|You tried to include by tagged region, but the included document does not have that tag.
+|<<include-partial>>
+
+|WARNING
+|<source> callout list item index: expected <x> got <y>
+|
+|<<callouts>>
+
+|WARNING
+|<source> include file not found: <file>
+|Probably a typo or missing file. If not, make sure you understand the search process.
+|<<include-directive-2>>, <<include-resolution>>
+
+|WARNING
+|<source> invalid empty <x> detected in style attribute
+|
+|
+
+|WARNING
+|<source> invalid style for <x> block: <y>
+|
+|
+
+|WARNING
+|<source> invalid style for paragraph: <x>
+|
+|<<style>>
+
+|WARNING
+|<source> list item index: expected <x>, got <y>
+|
+|<<ordered-lists>>
+
+|WARNING
+|<source> multiple ids detected in style attribute
+|
+|<<id>>
+
+|WARNING
+|<source> no callouts refer to list item <x>
+|
+|<<callouts>>
+
+|WARNING
+|<source> section title out of sequence
+|Invalid document structure. Check leveloffsets.
+|<<sections>>
+
+|====
+
+(1) Failures are fatal - no output document will be produced.
+
+Errors do not stop conversion, but the output document will almost certainly be wrong.
+
+Warnings do not stop conversion, but they indicate possible problems, and the output may not be what you were expecting.
+
+// Rare/unexplained
+
+////
+|WARNING
+|gem \'thread_safe\' is not installed. This gem is recommended when registering custom converters.
+|
+|
+
+|WARNING
+|gem \'thread_safe\' is not installed. This gem is recommended when using custom backend templates.
```

Comment:
This warning occurs if you use custom templates (`-T <template_dir>`) and you try to use asciidoctor without first installing the thread_safe gem.